src/win.rs

```diff
@@ -75,94 +83,41 @@ impl Endpoint {
 
     /// Make new connection using the provided path and running event pool.
     pub async fn connect<P: AsRef<Path>>(path: P) -> io::Result<Connection> {
-        Ok(Connection::wrap(Self::connect_inner(path.as_ref())?))
-    }
+        let path = path.as_ref();
+
+        let mut duration = Duration::new(0, 0);
+        let client = loop {
+            match named_pipe::ClientOptions::new()
+                .read(true)
+                .write(true)
+                .open(path)
+            {
+                Ok(client) => break client,
+                Err(e)
+                    if e.raw_os_error()
+                        == Some(winapi::shared::winerror::ERROR_PIPE_BUSY as i32) =>
+                {
+                    if duration >= PIPE_AVAILABILITY_TIMEOUT {
+                        return Err(e);
+                    }
+                }
+                Err(e) => return Err(e),
+            }
 
-    fn connect_inner(path: &Path) -> io::Result<NamedPipe> {
-        use std::fs::OpenOptions;
-        use std::os::windows::fs::OpenOptionsExt;
-        use std::os::windows::io::{FromRawHandle, IntoRawHandle};
-        use winapi::um::winbase::FILE_FLAG_OVERLAPPED;
+            let now = Instant::now();
+            tokio::time::sleep(Duration::from_millis(50)).await;
+            duration += now.elapsed();
```

Comment:
We do we only count the `sleep` time and not the `opening` time? Like I think the code could be simplified by creating an `Instant` instead of initial `duration` and then checking `start.elapsed()` instead of summing up here?

Reply:
AIUI calling `CreateFile` to open an overlapped i/o named pipe should return immediately, that's why I didn't bother counting since from before the call but I didn't think of just checking `start.elapsed()` as you're saying - that's a lot cleaner. Thanks! Will fix this in a sec.

Reply:
Done in https://github.com/NikVolf/parity-tokio-ipc/pull/32/commits/b86fd388febfd0c55d5d678b21fa2bb9c2ca05b6